oak/server/rust/oak_runtime/src/runtime/introspection_events.rs

```diff
@@ -0,0 +1,51 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+pub use crate::proto::oak::introspection_events::event::EventDetails;
+use crate::{proto::oak::introspection_events::Event, runtime::Runtime};
+use log::info;
+use prost_types::Timestamp;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+// Introspection event related methods for the Runtime.
+impl Runtime {
+  /// Generates an introspection event recording a modification to the Oak
+  /// application.
+  #[cfg(feature = "oak_debug")]
+  pub fn introspection_event(&self, event_details: EventDetails) {
+    let duration_since_unix_epoch = SystemTime::now()
+      .duration_since(UNIX_EPOCH)
+      .expect("Time went backwards");
+
+    let timestamp = Timestamp {
+      seconds: duration_since_unix_epoch.as_secs() as i64,
+      nanos: duration_since_unix_epoch.subsec_nanos() as i32,
+    };
```

Comment:
nit: be nice to put this in a function of its own, so you can just have
```Rust
let event = Event {
  timestamp: Some(timestamp_for_now()),
  event_details: Some(event_details),
};
```

Reply:
Good idea :)